src/scheduler/VirtualTimeScheduler.ts

```diff
@@ -24,7 +24,15 @@ export class VirtualTimeScheduler extends AsyncScheduler {
     const {actions, maxFrames} = this;
     let error: any, action: AsyncAction<any>;
 
-    while ((action = actions.shift()) && (this.frame = action.delay) <= maxFrames) {
+    while (actions.length > 0) {
```

Comment:
To fix delay, we shouldn't have done anything to the VirtualTimeScheduler, changes to the virtual scheduler will affect _all marble tests_.

Reply:
This change has nothing to do with the fix. It's just I wanted to test that the last action is correctly unsubscribed and when calling scheduler.flush() it removed the action that wasn't executed 
```
actions.length === 1
maxFrames === 60
action.delay === 80 
(action = actions.shift()) && (this.frame = action.delay) <= maxFrames
true && false
// no execution
actions.length === 0 // same as if action would execute
```